File: csharp/Udc.Services.InternalGrpcApi.FakeClient/Builders/FakeCenterBuilder.cs
using Udc.Services.InternalGrpcApi.FakeClient.Helpers;
using Udc.Services.Protos.Fake;

namespace Udc.Services.InternalGrpcApi.FakeClient.Builders;

public class FakeCenterBuilder
{
    private FakeBaseCenter _fakeBaseCenter;

    public string TestId => "614";
    public string TestCode => "614";
    public string TestName => "Faculty of Computer Science";

    public FakeCenterBuilder()
    {
        _fakeBaseCenter = WithDefaultValues();
    }

    public FakeBaseCenter Build()
    {
        return _fakeBaseCenter;
    }

    public FakeBaseCenter WithDefaultValues()
    {
        _fakeBaseCenter = new FakeBaseCenter()
        {
            FakeBaseEntity = new FakeBaseEntity()
            {
                Id = TestId
            },
            Code = TestCode,
            Name = TestName
        };

        return _fakeBaseCenter;
    }
}

File: java/udc.services.internalgrpcapi.fakeclient/src/main/java/udc/services/internalgrpcapi/fakeclient/builders/FakeCenterBuilder.java
package udc.services.internalgrpcapi.fakeclient.builders;

import com.google.protobuf.StringValue;

import udc.services.internalgrpcapi.fakeclient.protos.fake.Fake.FakeBaseCenter;
import udc.services.internalgrpcapi.fakeclient.protos.fake.Fake.FakeBaseEntity;;

public class FakeCenterBuilder {
    
	private FakeBaseCenter _fakeBaseCenter;
	
    public final String TestId = "614";
    public final String TestCode = "614";
    public final String TestName = "Faculty of Computer Science";

    public FakeCenterBuilder()
    {
        _fakeBaseCenter = withDefaultValues();
    }

    public FakeBaseCenter build()
    {
        return _fakeBaseCenter;
    }

    public FakeBaseCenter withDefaultValues()
    {
    	FakeBaseEntity fakeBaseEntity = FakeBaseEntity
    			.newBuilder()
    			.setId(StringValue.newBuilder().setValue(TestId))
    			.build();
    	 _fakeBaseCenter = FakeBaseCenter
        		.newBuilder()
        		.setFakeBaseEntity(fakeBaseEntity)
        		.setCode(TestCode)
        		.setName(TestName)
        		.build();
    	
        return _fakeBaseCenter;
    }
}
